### DISCUSSION TITLE
didOpen notification

### DISCUSSION THREAD

**Discussion by cdietrich** [eclipse-langium/langium#1214] [Q&A]
Hi, where and how does langium handle `didOpen` notification?
is it possible to have a set of completely transient documents?

**Comment by msujew** [comment]
@cdietrich We handle neither `didOpen` nor `didClose` events/notifications explicitly. We [only react to the `didChange` event](https://github.com/eclipse-langium/langium/blob/a4c7a58e9a79199d0f69cd3a9479c6ddfe99de21/packages/langium/src/lsp/language-server.ts#L226-L241) of the text documents. Note that the event is also fired when a file is opened in the editor.

> is it possible to have a set of completely transient documents?

I.e. documents that only live in a service instance for a short duration and are then deleted? We use something often in interpreters, see [here](https://github.com/TypeFox/langium-lox/blob/f28627cbe0ce9ce2d714a96a04f5ee3c95c53d14/src/interpreter/runner.ts#L14-L22) for example.
